ludwig/datasets/__init__.py

```diff
@@ -25,21 +25,47 @@ def _get_dataset_configs() -> Dict[str, DatasetConfig]:
     import importlib.resources
 
     config_files = [f for f in importlib.resources.contents(configs) if f.endswith(".yaml")]
-    config_objects = [load_dataset_config(f) for f in config_files]
+    config_objects = [_load_dataset_config(f) for f in config_files]
     return {c.name: c for c in config_objects}
 
 
-def get_dataset_config(dataset_name) -> DatasetConfig:
+def _get_dataset_config(dataset_name) -> DatasetConfig:
     """Get the config for a dataset."""
     configs = _get_dataset_configs()
     if dataset_name not in configs:
         raise AttributeError(f"No config found for dataset {dataset_name}")
     return configs[dataset_name]
 
 
+def _load_model_config(model_config_filename: str):
+    """Loads a model config."""
+    model_config_path = os.path.join(os.path.dirname(model_configs.__file__), model_config_filename)
+    with open(model_config_path) as f:
+        return yaml.safe_load(f)
+
+
+@lru_cache(maxsize=3)
+def _get_model_configs(dataset_name: str) -> Dict[str, Dict]:
```

Comment:
Can the scope of this be expanded to fetch the "best" model config, like the configs that @abidwael and @jppgks have been checking in?

e.g. https://github.com/ludwig-ai/ludwig/pull/2536

Reply:
Good idea!  Added another special model config name "best", similar to "default" - what do you think of this?

Reply:
Looks good to me. The key-lookup API seems nicely general enough to expand to other categories as well, if/when we introduce them.